data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/converter/RecordConverter.java

```diff
@@ -76,14 +77,16 @@ void flushBuffer() throws Exception {
      * @param keys                    A map to hold the keys (partition key and sort key)
      * @param eventCreationTimeMillis Creation timestamp of the event
      * @param eventName               Event name
+     * @param userIdentity            UserIdentity for TTL based deletes
      * @throws Exception Exception if failed to write to buffer.
      */
     public void addToBuffer(final AcknowledgementSet acknowledgementSet,
                             final Map<String, Object> data,
                             final Map<String, Object> keys,
                             final long eventCreationTimeMillis,
                             final long eventVersionNumber,
-                            final String eventName) throws Exception {
+                            final String eventName,
+                            final Boolean userIdentity) throws Exception {
```

Comment:
This boolean should named according to whether it's a TTL delete. Alternatively, it may make sense to make this a String and do the boolean conversion to TTL delete in this method